gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_sdk_custom_metric.ipynb

```diff
@@ -0,0 +1,1390 @@
+        "    eval_metrics_weights=[0.8, 0.2],\n",
+        "    aggregation_type=\"weighted_sum\",  # Supported aggregation types: \"weighted_sum\", \"weighted_average\"\n",
+        "    data_limit=50,\n",
+        "    response_mime_type=\"application/json\",\n",
+        "    language=\"English\",  # Supported languages: \"English\", \"French\", \"German\", \"Hebrew\", \"Hindi\", \"Japanese\", \"Korean\", \"Portuguese\", \"Simplified Chinese\", \"Spanish\", \"Traditional Chinese\"\n",
+        "    placeholder_to_content=json.loads(\"{}\"),\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Jd_uzQYQx6L7"
+      },
+      "source": [
+        "#### Upload Vertex AI Prompt Optimizer (Preview) config to Cloud Storage\n",
+        "\n",
+        "After you define Vertex AI Prompt Optimizer (Preview) configuration, you upload them on Cloud Storage bucket.\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "QCJAqcfWBqAh"
+      },
+      "source": [
+        "Now you can save the config to the bucket."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "iqiv8ApR_SAM"
+      },
+      "outputs": [],
+      "source": [
+        "args = vars(args)\n",
+        "\n",
+        "with epath.Path(CONFIG_FILE_URI).open(\"w\") as config_file:\n",
+        "    json.dump(args, config_file)\n",
+        "config_file.close()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "spqgBT8hYAle"
+      },
+      "source": [
+        "#### Run the automatic prompt optimization job\n",
+        "\n",
+        "Now you are ready to run your first Vertex AI Prompt Optimizer (Preview) job using the Vertex AI SDK for Python.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "GtPnvKIpUQ3q"
+      },
+      "outputs": [],
+      "source": [
+        "WORKER_POOL_SPECS = [\n",
+        "    {\n",
+        "        \"machine_spec\": {\n",
+        "            \"machine_type\": \"n1-standard-4\",\n",
+        "        },\n",
+        "        \"replica_count\": 1,\n",
+        "        \"container_spec\": {\n",
+        "            \"image_uri\": APD_CONTAINER_URI,\n",
+        "            \"args\": [\"--config=\" + CONFIG_FILE_URI],\n",
+        "        },\n",
+        "    }\n",
+        "]\n",
+        "\n",
```

Comment:
Is there a built in class to use instead of a dict?

Reply:
This comes directly from the [documentation](https://cloud.google.com/vertex-ai/docs/training/distributed-training#configure_a_distributed_training_job)